File: xmp/FileSender.js
#!/usr/bin/env node
if ( require.main === module )
{
  var gepard = require ( "gepard" ) ;

  if ( gepard.getProperty ( "help" ) )
  {
    console.log (
      "Gepard Examples: FileSender, send a file and get an ack.\n"
    + "Usage: node FileSender.js [--file=<file>], default: FileSender.js\n"
    ) ;
    process.exit() ;
  }

  new gepard.Admin().isRunning ( function admin_is_running ( state )
  {
    if ( ! state )
    {
      console.log ( "Not running on " + this.getHostPort() ) ;
			process.exit() ;
    }
   	var client = gepard.getClient() ;
    var event = new gepard.Event ( "__FILE__" ) ;
    var file = gepard.getProperty ( "file", "FileSender.js" ) ;
    event.putValue ( "DATA", new gepard.FileContainer ( file ) ) ;
    client.request ( event, function ( e )
    {
      if ( e.isBad() )
      {
        console.log ( e ) ;
      }
      else
      {
        console.log ( "File " + file + " sent successfully." )
        console.log ( e.getStatus() ) ;
      }
			this.end() ;
    }) ;
  });
}

File: python/xmp/FileSender.py
#!/usr/bin/env python

import os, sys
sys.path.insert ( 0, os.path.dirname(os.path.abspath(__file__) ) + "/../" )
import gepard

client = gepard.Client.getInstance()

event = gepard.Event ( "__FILE__" )

file = gepard.util.getProperty ( "file", "FileSender.py" )
event.putValue ( "DATA", gepard.FileContainer ( file ) )

def result ( e ):
	if e.isBad():
		print ( e.getStatus() )
	else:
		print ( e.getStatus() )
		print ( "File " + file + " sent successfully." )
	e.getClient().close()

print ( "Sending " + file )
client.request ( event, result )
